packages/sign-client/src/controllers/engine.ts

```diff
@@ -760,13 +783,19 @@ export class Engine extends IEngine {
 
   private registerExpirerEvents() {
     this.client.expirer.on(EXPIRER_EVENTS.expired, async (event: ExpirerTypes.Expiration) => {
-      const { topic } = event;
-      if (this.client.session.keys.includes(topic)) {
-        await this.deleteSession(topic);
-        this.client.events.emit("session_expire", { topic });
-      } else if (this.client.pairing.keys.includes(topic)) {
-        await this.deletePairing(topic);
-        this.client.events.emit("pairing_expire", { topic });
+      const [type, value] = event.target.split(":");
+      if (type === "topic" && typeof value === "string") {
+        const topic = value;
+        if (this.client.session.keys.includes(topic)) {
+          await this.deleteSession(topic);
+          this.client.events.emit("session_expire", { topic });
+        } else if (this.client.pairing.keys.includes(topic)) {
+          await this.deletePairing(topic);
+          this.client.events.emit("pairing_expire", { topic });
+        }
+      } else if (type === "id" && typeof value === "number") {
```

Comment:
Do we need to check the fall through case here (i.e. if it matched neither `if` nor `else if` for some reason)?